api/v1alpha1/backstage_types.go

```diff
@@ -32,13 +32,19 @@ type BackstageSpec struct {
 	RawRuntimeConfig RuntimeConfig `json:"rawRuntimeConfig,omitempty"`
 
 	// Control the creation of a local PostgreSQL DB. Set to false if using for example an external Database for Backstage.
-	// To use an external Database, you can provide your own app-config file (see the AppConfig field in the Application structure)
-	// containing references to the Database connection information,
-	// which might be supplied as environment variables (see the ExtraEnvs field) or extra-configuration files
-	// (see the ExtraFiles field in the Application structure).
 	// +optional
 	//+kubebuilder:default=true
 	EnableLocalDb *bool `json:"enableLocalDb,omitempty"`
+
+	// Reference to an existing secret for database access. Required for external database access.
+	// Optional for a local database (EnableLocalDb=true) and if absent a secret will be auto generated.
+	// The secret shall include information requried for the database access, for instance:
```

Comment:
```suggestion
	// The secret shall include information required for the database access, for instance:
```

Reply:
Also, all those env vars are required only if `enableLocalDb` is `true`, as those are required by the PostgreSQL image we are using.
Otherwise, IMO, there is nothing preventing users from overriding the default database settings with an app-config like [this](https://github.com/janus-idp/backstage-showcase/blob/main/app-config.production.yaml), and providing an existing DB secret with those keys:
```yaml
backend:
  database:
    connection:
      host: "${MY_HOST}"
     port: "${MY_PORT}"
     user: "${MY_USER}"
     password: "${MY_PASSWORD}"
```